### PR TITLE
GEOT-6295: fixed densification in StreamingRenderer for small bboxes

### PR DESCRIPTION
**Pull request by mbarto**

### CODE REVIEW

File: modules/library/referencing/src/main/java/org/geotools/referencing/operation/transform/WarpBuilder.java [hunk 1]
```diff
@@ -84,6 +84,26 @@ public void setMaxPositions(int maxPositions) {
...
+     * @return
+     */
+    public boolean isValidDomain(Rectangle2D.Double domain) {
+    	return isValidDomain(domain.getMinX(), domain.getMaxX(), domain.getMinY(), domain.getMaxY());
+    }
+    
+    private boolean isValidDomain(double minx, double maxx, double miny, double maxy) {
+    	final int width = (int) (maxx - minx);
+        final int height = (int) (maxy - miny);
+        if (abs(width) == 0 || height == 0) {
```
Comment (aaime) [inline modules/library/referencing/src/main/java/org/geotools/referencing/operation/transform/WarpBuilder.java]: Can be simpler, return directly:
``return abs(width) > 0 && abs(height)== 0``

File: modules/library/render/src/test/java/org/geotools/renderer/lite/StreamingRendererTest.java [hunk 2]
```diff
@@ -232,6 +232,43 @@ public void testDensification() throws Exception {
...
+        StyleBuilder sb = new StyleBuilder();
+        mc.addLayer(new FeatureLayer(zzSource, sb.createStyle(sb.createLineSymbolizer())));
+        StreamingRenderer sr = new StreamingRenderer();
+        Map hints = new HashMap();
+        hints.put(StreamingRenderer.ADVANCED_PROJECTION_HANDLING_KEY, true);
+        hints.put(StreamingRenderer.ADVANCED_PROJECTION_DENSIFICATION_KEY, true);
+        sr.setRendererHints(hints);
+        sr.setMapContent(mc);
+
+        /*BufferedImage bi = new BufferedImage(1, 1, BufferedImage.TYPE_3BYTE_BGR);
```
Comment (aaime) [inline modules/library/render/src/test/java/org/geotools/renderer/lite/StreamingRendererTest.java]: No commented out code please :-)

**Comment by mbarto** [comment]
@aaime should be ok now

**Comment by mbarto** [comment]
@aaime travis is passing now

**Comment by aaime** [comment]
Looks good to me

**Pull request merged**